src/View/Components/Imgix.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Astrotomic\Imgix\View\Components;
+
+use Astrotomic\Imgix\ImgixManager;
+use Illuminate\View\Component;
+
+class Imgix extends Component
+{
+    protected ImgixManager $imgixManager;
+    protected string $path;
+
+    protected ?string $source;
+    protected ?string $imgWidth;
+    protected ?string $imgHeight;
+
+    public function __construct(
+        ImgixManager $imgixManager,
+        string $path,
+        ?string $source = null,
+        ?string $width = null,
+        ?string $height = null
+    )
```

Comment:
The `params` are missing here
> optional `params` which will be passed to the imgix methods

This would be needed to define crop mode, color overlay or whatever you want to use by the imgix API.
They should be

Reply:
So attributes are mixed html attributes like `class` or `alt` _and_ imgix attributes like [`crop`](https://docs.imgix.com/apis/rendering/size/crop) or [`fit`](https://docs.imgix.com/apis/rendering/size/fit)?

Reply:
The `$attributes` in the blade view contains all attributes passed to the blade component that aren't part of this construct.
What I would like is a `array $params` like you define in the `src()` or `ssrcSet()` methods which allows the user to pass in his own parameters like `<x-imgix path="image.jpg" :params="['crop' => 'edges', 'fit' => 'crop']"/>`

Reply:
The other one WOULD be possible but I think it's too much work for the moment. So a component like `<x-imgix crop="edges" />` would be cool but I think it requires an allow/deny list for valid `<img/>` HTML attributes and/or Imgix API params.

Reply:
Added the `:params` solutions for now.

Reply:
perfect. that's what I wanted. 🙂